src/Controller/MainController.php

```diff
@@ -37,13 +37,35 @@ public function __construct(\Twig_Environment $twig, EntityRepository $threadRep
      */
     public function indexAction(Request $request)
     {
-        $response = new Response();
-        $threads = $this->threadRepository->findAll();
+
+
+        //$allThreads = count( $this->threadRepository -> findAll());
+        $currentPage = $request->get('page', 1);
+
+        //var_dump(($currentPage - 1) * 10); die;
+
+        $totalCount = $this->threadRepository->createQueryBuilder('u')
+            ->select('count(u.id)')
```

Comment:
the "u" character that is used by the query builder here is the alias for the object so that you can reference it in other functions like the select. Being picky here but let's use "t" as the alias instead of "u". The convention is first letter of the class. Also it is then consistent with anywhere else we build thread queries in the application which is useful.